packages/transport/src/lowlevel/sharedConnectionWorker.ts

```diff
@@ -223,37 +162,92 @@ function handleAcquireIntent(
-        sendBack({ type: `wrong-previous-session` }, id, port);
+        sendBack({ type: 'wrong-previous-session' }, id, port);
         return Promise.resolve();
     }
     startLock();
-    sendBack({ type: `other-session`, otherSession: otherTable[path] }, id, port);
+    sendBack({ type: 'other-session', otherSession: otherTable[path] }, id, port);
     // if lock times out, promise rejects and queue goes on
-    // @ts-ignore
-    return waitForLock().then((obj: { good: boolean; id: number }) => {
+    return waitForLock().then(obj => {
         if (obj.good) {
             lastSession++;
             let session = lastSession.toString();
             if (debug) {
                 session = `debug${session}`;
             }
             thisTable[path] = session;
-            sendBack({ type: `session-number`, number: session }, obj.id, port);
+            sendBack({ type: 'session-number', number: session }, obj.id, port);
         } else {
             // failure => nothing happens, but still has to reply "ok"
-            sendBack({ type: `ok` }, obj.id, port);
+            sendBack({ type: 'ok' }, obj.id, port);
         }
     });
 }
 
-function sendBack(message: MessageFromSharedWorker, id: number, port: PortObject) {
-    port.postMessage({ id, message });
+function handleMessage(
+    { id, message }: { id: number; message: MessageToSharedWorker },
+    port: PortObject,
+) {
+    if (message.type === 'acquire-intent') {
+        const { path } = message;
+        const { previous } = message;
+        const { debug } = message;
+        waitInQueue(() => handleAcquireIntent(path, id, port, previous, debug));
+    }
+    if (message.type === 'acquire-done') {
+        handleAcquireDone(id); // port is the same as original
+    }
+    if (message.type === 'acquire-failed') {
+        handleAcquireFailed(id); // port is the same as original
+    }
+    if (message.type === 'get-sessions') {
+        waitInQueue(() => handleGetSessions(id, port));
+    }
+
+    if (message.type === 'get-sessions-and-disconnect') {
+        const { devices } = message;
+        waitInQueue(() => handleGetSessions(id, port, devices));
+    }
+
+    if (message.type === 'release-onclose') {
+        const { session } = message;
+        waitInQueue(() => handleReleaseOnClose(session));
+    }
+
+    if (message.type === 'release-intent') {
+        const { session } = message;
+        const { debug } = message;
+        waitInQueue(() => handleReleaseIntent(session, debug, id, port));
+    }
+    if (message.type === 'release-done') {
+        handleReleaseDone(id); // port is the same as original
+    }
+    if (message.type === 'enumerate-intent') {
+        waitInQueue(() => handleEnumerateIntent(id, port));
+    }
+    if (message.type === 'enumerate-done') {
+        handleReleaseDone(id); // port is the same as original
+    }
+}
+
+// TODO:
+// proper ts configuration for lib webworker
+// @ts-ignore
+if (typeof onconnect !== 'undefined') {
```

Comment:
not sure, but when you will have correct types for webworker in tsconfig, then you will probably still need to do `self.onconnect`

Reply:
Here I decided to go with @ts-ignore for now as worker types were clashing with DOM types and I didnt find a quick way around it.

Reply:
Ok, we can solve this later